test/common_utils.py

```diff
@@ -270,84 +270,57 @@ def combinations_grid(**kwargs):
     return [dict(zip(kwargs.keys(), values)) for values in itertools.product(*kwargs.values())]
 
 
-class ImagePair(TensorLikePair):
-    def __init__(
-        self,
-        actual,
-        expected,
-        *,
-        mae=False,
-        **other_parameters,
-    ):
-        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
-            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
+assert_equal = functools.partial(torch.testing.assert_close, rtol=0, atol=0)
 
-        super().__init__(actual, expected, **other_parameters)
-        self.mae = mae
 
-    def compare(self) -> None:
-        actual, expected = self.actual, self.expected
+def assert_close_with_image_support(actual, expected, *, mae=False, atol=None, msg=None, **kwargs):
+    def compare(actual, expected):
+        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
```

Comment:
Why not convert all inputs to tensors, regardless of whether they're all PIL Images? Does that mean we can't do `assert_close_with_image_support(some_tensor, some_pil_img)`?

Reply:
See https://github.com/pytorch/vision/pull/7525#discussion_r1172776763 for one reason. Plus, we originally had the option to pass in one tensor and one PIL image and let the function handle the conversion to tensor. However, we often saw extra flakiness in the tests just to this conversion and thus we scraped that feature. Since this irregular anyway, the few tests that need this, perform the conversion themselves now. E.g.

https://github.com/pytorch/vision/blob/55799959046fe38e5bd324631e33238a6f8f08a7/test/transforms_v2_kernel_infos.py#L117